packages/web/pages/stake.tsx

```diff
@@ -279,13 +281,45 @@ export const Staking: React.FC = observer(() => {
     unstakeCall,
   ]);
 
+  const fetchAprData = async () => {
+    const response = await axios.get(
+      "https://public-osmosis-api.numia.xyz/apr?start_date=2023-12-29&end_date=2024-02-05",
+      {
+        headers: {
+          Authorization: `Bearer ${process.env.NEXT_PUBLIC_NUMIA_API_KEY}`,
+        },
+      }
+    );
+
+    console.log("response: ", response);
+    return response.data;
+  };
+
+  const getAverageApr = (data: { labels: string; apr: number }[] = []) => {
+    if (data.length === 0) return 0;
+
+    const sum = data.reduce((acc, item) => acc + item.apr, 0);
+    const average = sum / data.length;
+    return average;
+  };
+
+  const formatAverageApr = (apr: number) => {
+    return new Dec(apr);
+  };
+
+  const { data } = useQuery(["2022-01-05"], fetchAprData);
```

Comment:
We should def use TRPC imo but it would be better to merge working code and refactor after we releae staking if it's going to take another day to refactor

Reply:
I'm personally not opposed to direct use of react query in simple cases, but with tRPC we get:
* Ability to do fine-grained caching on our own on server (can prevent some calls to data services, but be careful not to return stale important data)
* Process and then cache the data that was processed on server
* Use more advanced types in the query hook and response (Dec, CoinPretty, etc)
* Can change the API to a query (I did this for the historical price query, [see](https://github.com/osmosis-labs/osmosis-frontend/blob/a85257f566e50d6e79dcdd034964b6747ec90a15/packages/web/server/api/edge-routers/assets-router.ts#L214))
* Type safe query invalidation instead of using string IDs (to help with this we wrapped the react query call in our [own hook](https://github.com/osmosis-labs/osmosis-frontend/blob/a85257f566e50d6e79dcdd034964b6747ec90a15/packages/web/hooks/queries/cosmos/use-balances.ts#L7))

Reply:
Though also once you get the hang of tRPC it takes no more than 5 minutes to add a new tRPC call, which is why it's so powerful and why you shouldn't hesitate to use it if any of the above benefits hold.